File: MiniDOMOffice/CRM/Interazioni/Telefonate/CTelefonatePersona.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases

Imports minidom.Anagrafica

Partial Public Class CustomerCalls


    ''' <summary>
    ''' rappresenta le telefonate fatte ad una persona
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CTelefonatePersona
        Inherits CTelefonateCollection

        Private m_Persona As CPersona

        Public Sub New()
            Me.m_Persona = Nothing
        End Sub
        Public Sub New(ByVal persona As CPersona)
            Me.New()
            Me.Load(persona)
        End Sub

        Protected Overrides Sub OnInsert(index As Integer, value As Object)
            If (Me.m_Persona IsNot Nothing) Then DirectCast(value, CTelefonata).Persona = Me.m_Persona
            MyBase.OnInsert(index, value)
        End Sub

        Protected Friend Sub Load(ByVal persona As CPersona)
            If (persona Is Nothing) Then Throw New ArgumentNullException("Persona")
            MyBase.Clear()
            Me.m_Persona = persona
            If (GetID(persona) = 0) Then Exit Sub
            Dim cursor As New CCustomerCallsCursor
            Try
                cursor.IDPersona.Value = GetID(persona)
                cursor.Data.SortOrder = SortEnum.SORT_DESC
                cursor.Stato.Value = ObjectStatus.OBJECT_VALID
                cursor.ClassName.Value = "CTelefonata"
                cursor.ClassName.Operator = OP.OP_LIKE
                cursor.IgnoreRights = True
                While Not cursor.EOF
                    MyBase.Add(cursor.Item)
                    cursor.MoveNext()
                End While
            Catch ex As Exception
                Throw
            Finally
                If (cursor IsNot Nothing) Then cursor.Dispose() : cursor = Nothing
            End Try
        End Sub


    End Class



End Class

File: MiniDOMOffice/CRM/Interazioni/Telefonate/Telefonate/CTelefonatePersona.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.CustomerCalls;

namespace minidom
{
    public partial class CustomerCalls
    {


        /// <summary>
        /// rappresenta le telefonate fatte ad una persona
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CTelefonatePersona 
            : CCollection<CTelefonata>
        {
            
            [NonSerialized] private Anagrafica.CPersona m_Persona;

            /// <summary>
            /// Costruttore
            /// </summary>
            public CTelefonatePersona()
            {
                m_Persona = null;
            }

            /// <summary>
            /// Costruttore
            /// </summary>
            /// <param name="persona"></param>
            public CTelefonatePersona(CPersona persona) : this()
            {
                if (persona is null)
                    throw new ArgumentNullException("persona");

                Load(persona);
            }

            /// <summary>
            /// OnInsert
            /// </summary>
            /// <param name="index"></param>
            /// <param name="value"></param>
            protected override void OnInsert(int index, CTelefonata value)
            {
                if (m_Persona is object)
                    value.SetPersona(this.m_Persona);
                base.OnInsert(index, value);
            }

            /// <summary>
            /// OnSet
            /// </summary>
            /// <param name="index"></param>
            /// <param name="oldValue"></param>
            /// <param name="newValue"></param>
            protected override void OnSet(int index, CTelefonata oldValue, CTelefonata newValue)
            {
                if (this.m_Persona is object)
                    newValue.SetPersona(this.m_Persona);
                base.OnSet(index, oldValue, newValue);
            }

            /// <summary>
            /// Carica dal db
            /// </summary>
            /// <param name="persona"></param>
            protected internal void Load(Anagrafica.CPersona persona)
            {
                if (persona is null)
                    throw new ArgumentNullException("Persona");
                
                this.Clear();

                this.m_Persona = persona;

                if (DBUtils.GetID(persona, 0) == 0)
                    return;

                using (var cursor = new CContattoUtenteCursor())
                {
                    cursor.IDPersona.Value = DBUtils.GetID(persona, 0);
                    cursor.Data.SortOrder = SortEnum.SORT_DESC;
                    cursor.Stato.Value = ObjectStatus.OBJECT_VALID;
                    cursor.ClassName.Value = "CTelefonata";
                    cursor.ClassName.Operator = OP.OP_LIKE;
                    cursor.IgnoreRights = true;
                    while (cursor.Read())
                    {
                        Add((CTelefonata)cursor.Item);
                    }
                }
                     
            }
        }
    }
}